### PR TITLE
PredictiveCacheController: - refactored constructor; - added zoom level options for maps predictive cache options

### PR DESCRIPTION
**Pull request by RingerJK**
### Description
see title

### CODE REVIEW

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt [hunk 1]
```diff
@@ -332,8 +345,14 @@ class PredictiveCacheController @JvmOverloads constructor(
...
                 .styleURI(styleURI)
-                .minZoom(MIN_ZOOM)
-                .maxZoom(MAX_ZOOM)
+                .minZoom(
+                    predictiveCacheOptions.predictiveCacheMapsOptions
+                        .predictiveCacheZoomLevelOptions.minZoom
+                )
+                .maxZoom(
+                    predictiveCacheOptions.predictiveCacheMapsOptions
+                        .predictiveCacheZoomLevelOptions.minZoom
```
Comment (dzinad) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt:354]: maxZoom
There should be unit tests that would fail...

Comment (RingerJK) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt:354]: great catch 👍

**Review by dzinad — APPROVED** [review]

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/options/PredictiveCacheZoomLevelOptions.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package com.mapbox.navigation.base.options
+
+/**
+ * Maps predictive cache zoom level options.
+ *
+ * @param minZoom Minimum zoom level for the tile package. See **com.mapbox.maps.TilesetDescriptorOptions#getMinZoom**
+ * @param maxZoom Maximum zoom level for the tile package. See **com.mapbox.maps.TilesetDescriptorOptions#getMaxZoom**
+ */
+class PredictiveCacheZoomLevelOptions private constructor(
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/options/PredictiveCacheZoomLevelOptions.kt:9]: Do we need a separate class? It feels like these could just be part of the map options.

**Review by LukasPaczos — COMMENTED** [review]
Did you give this feature a test run as well?

We should set some narrow zoom level range, set a route and simulate navigation for a moment (without moving the map camera manually), disable internet connection, and verify that only tiles within the range have been cached along the route.

You can see the tile's zoom level with
```
mapboxMap.setDebug(listOf(MapDebugOptions.TILE_BORDERS, MapDebugOptions.PARSE_STATUS), true)
```

**Comment by RingerJK** [comment]
> Did you give this feature a test run as well?
> 
> We should set some narrow zoom level range, set a route and simulate navigation for a moment (without moving the map camera manually), disable internet connection, and verify that only tiles within the range have been cached along the route.
> 
> You can see the tile's zoom level with
> 
> ```
> mapboxMap.setDebug(listOf(MapDebugOptions.TILE_BORDERS, MapDebugOptions.PARSE_STATUS), true)
> ```

I made a few ground tests with Route Replay. Initially maps tiles are downloaded a pretty wide area around a current location, so the destination must be far away from a started (puck) location. It works 

The internet connection was disabled a few mins after navigation was started:

https://user-images.githubusercontent.com/8738058/180040058-2e50fee4-1f36-439e-a7fe-f1313edc42ef.mp4

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,10 @@ Mapbox welcomes participation and contributions from everyone.
...
 - Improved off-road detection for Auto profile by relying more on the heading change relative to the road. [#6061](https://github.com/mapbox/mapbox-navigation-android/pull/6061)
+- Added `PredictiveCacheController` constructor with `PredictiveCacheOptions`, `PredictiveCacheController(PredictiveCacheOptions)`. [#6071](https://github.com/mapbox/mapbox-navigation-android/pull/6071)
+- Deprecated constructors `PredictiveCacheController(PredictiveCacheLocationOptions, PredictiveCacheControllerErrorHandler)` and `PredictiveCacheController(PredictiveCacheLocationOptions, PredictiveCacheLocationOptions, PredictiveCacheControllerErrorHandler)`. [#6071](https://github.com/mapbox/mapbox-navigation-android/pull/6071)
+- Added `PredictiveCacheController#predictiveCacheControllerErrorHandler` to set and get `PredictiveCacheControllerErrorHandler`. [#6071](https://github.com/mapbox/mapbox-navigation-android/pull/6071)
+- Added `PredictiveCacheMapsOptions`(maps specific, include set upping custom zoom level for Predictive Cache Maps' tiles) and `PredictiveCacheNavigationOptions`(navigation specific) Predictive Cache Options. [#6071](https://github.com/mapbox/mapbox-navigation-android/pull/6071)
```
Comment (LukasPaczos) [inline CHANGELOG.md:20]:
```suggestion
- Added `PredictiveCacheMapsOptions`(map specific, that also allow to specify zoom levels for which the map tiles should be cached) and `PredictiveCacheNavigationOptions`(navigation specific) available through the `PredictiveCacheOptions`. [#6071](https://github.com/mapbox/mapbox-navigation-android/pull/6071)
```

**Pull request merged**